base_layer/core/src/validation/block_validators.rs

```diff
@@ -148,6 +150,22 @@ fn check_accounting_balance(
         })
 }
 
+fn check_block_weight(block: &Block, consensus_constants: &ConsensusConstants) -> Result<(), ValidationError> {
+    trace!(
+        target: LOG_TARGET,
+        "Checking weight of block with hash {}",
+        block.hash().to_hex()
+    );
+    // The genesis block has a larger weight than other blocks may have so we have to exclude it here
+    if block.body.calculate_weight() <= consensus_constants.get_max_block_transaction_weight() ||
```

Comment:
Swap these. `block.header.height == 0` is super quick to evaluate and can shortcut

Reply:
true, but its only going to need that once, at height 0. If its any other height and it passes the calculate height it should not even bother to check height == 0